tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/core/util/ResourceUtilsTest.java

```diff
@@ -465,5 +467,56 @@ public void testGetRouteForService() {
 		assertThat(route).isNull();
 	}
 
+	@Test
+	public void testGetDeploymentConfigNameForPods() {
+		// given
+		// when
+		String name = getDeploymentConfigNameForPods(null);
+		// then
+		assertThat(name).isNull();
+
+		// given
+		// when
+		name = getDeploymentConfigNameForPods(Collections.emptyList());
+		// then
+		assertThat(name).isNull();
+
+		// given
+		List<IPod> pods = Arrays.asList(ResourceMocks.createResource(IPod.class),
+				pod,
+				ResourceMocks.createResource(IPod.class));
+		// when
+		name = getDeploymentConfigNameForPods(pods);
+		// then
+		assertThat(name).isNull();
+
+		// given
+		final HashMap<String, String> podLabels = new HashMap<>();
+		podLabels.put("foo", "booh");
+		podLabels.put("bar", "car");
+		podLabels.put(ResourceUtils.DEPLOYMENT_CONFIG_KEY, "hooolahoo");
+		IPod pod = ResourceMocks.createResource(IPod.class, p -> when(p.getLabels()).thenReturn(podLabels));
+		pods = Arrays.asList(ResourceMocks.createResource(IPod.class),
+				pod,
+				ResourceMocks.createResource(IPod.class));
+		// when
+		name = getDeploymentConfigNameForPods(pods);
+		// then
+		assertThat(name).isEqualTo("hooolahoo");
+
+		// given
+		final HashMap<String, String> otherPodLabels = new HashMap<>();
+		podLabels.put(ResourceUtils.DEPLOYMENT_CONFIG_KEY, "hookaboo");
+		podLabels.put(ResourceUtils.DEPLOYMENT_CONFIG_KEY, "hooolahoo");
+		pod = ResourceMocks.createResource(IPod.class, p -> when(p.getLabels()).thenReturn(otherPodLabels));		
+		pods = Arrays.asList(ResourceMocks.createResource(IPod.class),
+				pod,
+				ResourceMocks.createResource(IPod.class));
+		// when
+		name = getDeploymentConfigNameForPods(pods);
+		// then
+		assertThat(name).isEqualTo("hookaboo");
```

Comment:
Failed tests:
  ResourceUtilsTest.testGetDeploymentConfigNameForPods:518
Expecting:
 <null>
to be equal to:
 <"hookaboo">
but was not.